cowprotocol/cowswap
# 332/quote id on metadata

# Summary

Part of #332 

⚠️  *Note:* Build will fail until this change is merged and published https://github.com/cowprotocol/cow-sdk/pull/39

Passing down quoteId received from quote call, if any

  # To Test

Almost same steps as https://github.com/cowprotocol/cowswap/pull/629

1. Fill in trade data (or do nothing, as it'll pre-fill with WETH-USDC pair)
2. Observe the console logs. Filter by `appData`
* It should contain `id` inside the `quote` object
![Screen Shot 2022-06-27 at 16 47 08](https://user-images.githubusercontent.com/43217/175980690-a5cb5771-70b0-4009-b4c9-11f2a2df84b9.png)

4. Place an order and open it on the browser `https://barn.api.cow.fi/mainnet/api/v1/orders/<orderId>`
* Check the appDataHash in the order matches the last console debug log
5. Filter the console log for `UploadToIpfs`
* Check the appData related to your order has been uploaded, like the image
<img width="475" alt="Screen Shot 2022-06-08 at 15 10 57" src="https://user-images.githubusercontent.com/43217/172638536-5634007f-d9fe-4a76-b77c-c624bec0d6ba.png">

7. Search for the appDataHash using https://codepen.io/pacara/pen/BadZgjg?editors=0101
* Check it exists and the fields match the last console debug log


# Note
